lib/matplotlib/image.py

```diff
@@ -13,6 +13,7 @@
 
 from math import ceil
 import os
+import warnings
```

Comment:
I'd strongly suggest this is a use for `logging` instead of warnings.

```
import logging
....

_log = logging.getLogger(__name__)

....
_log.warn('Clipping...')
```

Reply:
I disagree, but I'm happy to let @tacaswell or another maintainer settle it.

FWIW I did a search for each pattern and found 21 log calls (mostly for import errors or configuration state changes) and 66 warnings (mostly for invalid calls or data), so warnings seem to be the convention here (as in Numpy, Pandas, etc).

Reply:
Logging was just added a couple of months ago.

Reply:
Just to follow up, when we added logging, we decided to follow the guidelines at https://docs.python.org/3/howto/logging.html#logging-basic-tutorial for when to use which tool:

> warnings.warn() in library code if the issue is avoidable and the client application should be 
> modified to eliminate the warning
>
> logging.warning() if there is nothing the client application can do about the situation, but the 
> event should still be noted

We decided *not* go back and check that the current instances of `warnings.warn` follow that pattern, but agreed to fix them as they come along.  

I'd drop my "strongly" above, and mildly argue that this warning is the latter case, and the particular advantage of `logging` is that the warnings can be turned off if the user is OK and cognizant w/ the clipping.  I don't think `warnings.warn` can be turned off, or at least not on a  per-module basis.

Reply:
Ah, that makes sense!  The [contributing guide is pretty good](https://matplotlib.org/devdocs/devel/contributing.html#which-logging-level-to-use), but I'll add a note with this explanation 😄